File: selenium/src/java/com/thoughtworks/selenium/internal/RegExTextMatchingStrategy.java
package com.thoughtworks.selenium.internal;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class RegExTextMatchingStrategy implements TextMatchingStrategy {
    public boolean isAMatch(String compareThis, String with) {
        Pattern pattern = Pattern.compile(compareThis, Pattern.MULTILINE);
        Matcher matcher = pattern.matcher(with);
        return matcher.find();

    }
}

File: selenium/src/csharp/WebdriverBackedSelenium/Internal/SeleniumEmulation/TextMatchingStrategies/RegExTextMatchingStrategy.cs
using System.Text.RegularExpressions;

namespace Selenium.Internal.SeleniumEmulation
{
    internal class RegexTextMatchingStrategy : ITextMatchingStrategy
    {
        public bool IsAMatch(string compareThis, string compareTo)
        {
            var pattern = new Regex(compareThis, RegexOptions.Multiline);
            Match matcher = pattern.Match(compareTo);
            return matcher.Success;
        }
    }
}